PowershellTools.HostService/ServiceManagement/ServiceCommon.cs

```diff
@@ -0,0 +1,136 @@
+        /// </summary>
+        /// <param name="msg"></param>
+        public static void Log(string msg)
+        {
+            Log(msg, ConsoleColor.Green);
+        }
+
+        /// <summary>
+        /// TODO: Temporary logging before having logging infrastructure ready
+        /// </summary>
+        /// <param name="msg"></param>
+        /// <param name="c"></param>
+        public static void Log(string msg, ConsoleColor c)
+        {
+            Console.ForegroundColor = c;
+            Console.WriteLine(msg);
+            Console.ResetColor();
+        }
+
+        #region private helpers
+
+        private void InitializeRunspace(PSHost psHost)
+        {
+            Log("Initializing run space with debugger", ConsoleColor.Green);
+            InitialSessionState iss = InitialSessionState.CreateDefault();
+            iss.ApartmentState = ApartmentState.STA;
+            iss.ThreadOptions = PSThreadOptions.ReuseThread;
+
+            _runspace = RunspaceFactory.CreateRunspace(psHost, iss);
+            _runspace.Open();
+
+            ImportPoshToolsModule();
+            LoadProfile();
+
+            SetupExecutionPolicy();
+        }
+
+        private void ImportPoshToolsModule()
+        {
+            using (PowerShell ps = PowerShell.Create())
+            {
+                try
+                {
+                    var assemblyLocation = Assembly.GetExecutingAssembly().Location;
+                    ps.Runspace = _runspace;
+                    ps.AddScript("Import-Module '" + assemblyLocation + "'");
+                    ps.Invoke();
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("Failed to load profile.", ex);
+                }
+            }
+        }
+
+        private void LoadProfile()
+        {
+            using (PowerShell ps = PowerShell.Create())
+            {
+                try
+                {
+                    var myDocuments = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
+                    var windowsPowerShell = Path.Combine(myDocuments, "WindowsPowerShell");
+                    var profile = Path.Combine(windowsPowerShell, "PoshTools_profile.ps1");
+
+                    var fi = new FileInfo(profile);
+                    if (!fi.Exists)
+                    {
+                        return;
+                    }
+
+                    ps.Runspace = _runspace;
+                    ps.AddScript(". '" + profile + "'");
+                    ps.Invoke();
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("Failed to load profile.", ex);
+                }
+            }
+        }
+
+        private void SetupExecutionPolicy()
+        {
+            SetExecutionPolicy(ExecutionPolicy.RemoteSigned, ExecutionPolicyScope.Process);
+        }
+
+        private void SetExecutionPolicy(ExecutionPolicy policy, ExecutionPolicyScope scope)

```

Comment:
Oops… fixed.

From: Andre Sayre [mailto:notifications@github.com]
Sent: Tuesday, January 27, 2015 4:54 PM
To: Microsoft/poshtools
Cc: Eric Deng
Subject: Re: [poshtools] Outproc(64bit) local debugging (#62)


In PowershellTools.HostService/ServiceManagement/ServiceCommon.cs<https://github.com/Microsoft/poshtools/pull/62#discussion_r23657859>:

> +                    ps.AddScript(". '" + profile + "'");

> +                    ps.Invoke();

> +                }

> +                catch (Exception ex)

> +                {

> +                    Console.WriteLine("Failed to load profile.", ex);

> +                }

> +            }

> +        }

> +

> +        private void SetupExecutionPolicy()

> +        {

> +            SetExecutionPolicy(ExecutionPolicy.RemoteSigned, ExecutionPolicyScope.Process);

> +        }

> +

> +        private void SetExecutionPolicy(ExecutionPolicy policy, ExecutionPolicyScope scope)

This doesn't include the other changes here recently made, eg, -force

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/poshtools/pull/62/files#r23657859>.